drivers/bluetooth/hci/userchan.c

```diff
@@ -93,24 +93,70 @@ static bool is_hci_event_discardable(const uint8_t *evt_data)
 	}
 }
 
-static struct net_buf *get_rx(const uint8_t *buf)
+static struct net_buf *get_rx_evt(const uint8_t *data)
 {
-	bool discardable = false;
-	k_timeout_t timeout = K_FOREVER;
+	bool discardable = is_hci_event_discardable(data);
+	struct net_buf *buf;
+
+	do {
+		buf = bt_buf_get_evt(data[1], discardable, discardable ? K_NO_WAIT : K_SECONDS(10));
```

Comment:
Suggest to add comment or extract `data[1]` to a variable with a informative name to improve readability